python/lsst/cp/pipe/cpFunctions.py

```diff
@@ -0,0 +1,516 @@
+                                                                instrument=self.instrument,
+                                                                calibration_label=self.calibrationLabel))
+            if len(existingValues) != 0:
+                raise RuntimeError(f"Existing entries for calibration_label {self.calibrationLabel}")
+        except LookupError:
+            pass
+
+        # Prepare combination of new data ids and object data:
+        self.newDataId = []
+        self.objects = []
+        for input in self.inputValues:
+            self.newDataId.append(input.dataId)
+            self.objects.append(self.butler.get(input))
+
+    def registerCalibrations(self, datasetTypeName):
+        """Add blessed inputs to the output collection.
+
+        Parameters
+        ----------
+        datasetTypeName : `str`
+            Dataset type these calibrations will be registered for.
+        """
+
+        # Find/make the run we will use for the output
+        run = self.registry.getRun(collection=self.outputCollection)
+        if run is None:
+            run = self.registry.makeRun(self.outputCollection)
+        self.butler.run = run
+        self.butler.collection = None
+
+        for newId, data in zip(self.newDataId, self.objects):
+            self.butler.put(data, datasetTypeName, dataId=newId,
+                            calibration_label=self.calibrationLabel,
+                            producer=None)
+
+    def addCalibrationLabel(self, name=None, instrument=None,
+                            beginDate="1970-01-01", endDate="2038-12-31"):
+        """Method to allow tasks to add calibration_label for master calibrations.
+
+        Parameters
+        ----------
+        name : `str`
+            A unique string for the calibration_label key.
+        instrument : `str`
+            Instrument this calibration is for.
+        beginDate : `str`
+            A hyphen delineated date string for the beginning of the valid date range.
+        endDate : `str`
+            A hyphen delineated date string for the end of the valid date range.
+        """
+        if name is None:
+            name = self.calibrationLabel
+        if instrument is None:
+            instrument = self.instrument
+
+        beginY, beginM, beginD = beginDate.split("-")
+        endY, endM, endD = endDate.split("-")
+
+        self.butler.registry.insertDimensionData(
+            "calibration_label",
+            {
+                "name": name,
+                "instrument": instrument,
+                "datetime_begin": datetime.datetime(int(beginY), int(beginM), int(beginD), 0, 0, 0),
+                "datetime_end": datetime.datetime(int(endY), int(endM), int(endD), 0, 0, 0)
+            }
+        )
+
+
+class CalibStatsConfig(pexConfig.Config):
+    """Parameters controlling the measurement of background statistics.
+    """
+    stat = pexConfig.Field(
+        dtype=int,
+        default=int(afwMath.MEANCLIP),
```

Comment:
I'm not sure this was ever a good idea, but it's my fault.

Reply:
I'm happy to think of the best option as part of a future ticket.  This should simply match existing code for the generation switch.